pkg/controller/image-stream-tag-reconciler/prowjobreconciler/prowjobreconciler.go

```diff
@@ -0,0 +1,249 @@
+package prowjobreconciler
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	prowv1 "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/kube"
+	"k8s.io/test-infra/prow/pjutil"
+	"sigs.k8s.io/controller-runtime"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	cioperatorapi "github.com/openshift/ci-tools/pkg/api"
+)
+
+// OrgRepoBranchCommit represents a GitHub commit
+type OrgRepoBranchCommit struct {
+	Org    string
+	Repo   string
+	Branch string
+	Commit string
+}
+
+// Enqueuer allows the caller to Enqueue an OrgRepoBranchCommit
+type Enqueuer func(OrgRepoBranchCommit)
+
+const controllerName = "promotion_job_creator"
+
+func AddToManager(mgr controllerruntime.Manager, config config.Getter, dryRun bool) (Enqueuer, error) {
+	createdJobsCounter := prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: controllerName,
+		Name:      "prowjobs_created",
```

Comment:
Can we call this something specific to this controller?

Reply:
it has the `Namespace` set to the `controllerName` so that is the case already

Reply:
AH